src/runez/http.py

```diff
@@ -49,6 +50,124 @@ def urljoin(base, url):
     return urllib.parse.urljoin(base, url, allow_fragments=False)
 
 
+class CacheWrapper:
```

Comment:
Could we use something like https://requests-cache.readthedocs.io/en/stable/ instead of implementing caching on the side?

Reply:
I will check this out and give it a try, didn't know it existed! Last time I searched for these, didn't find any good one back then. This one looks very neat!